pytorch_lightning/trainer/callback_hook.py

```diff
@@ -293,10 +293,22 @@ def on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> Dict[Type, dict]:
 
     def on_load_checkpoint(self, checkpoint):
         """Called when loading a model checkpoint."""
-        callback_states = checkpoint.get('callbacks')
+
         # Todo: the `callback_states` are dropped with TPUSpawn as they
         # can't be saved using `xm.save`
         # https://github.com/pytorch/xla/issues/2773
+        callback_states = checkpoint.get('callbacks')
+
+        current_callbacks_type = {type(cb) for cb in self.callbacks}
+        saved_callbacks_type = set(callback_states)
+        difference = saved_callbacks_type.difference(current_callbacks_type)
+        if difference:
+            rank_zero_warn(
+                "Be aware that when using ``resume_from_checkpoint``, "
+                "callbacks used to create the checkpoint need to be provided. "
+                f"Please, add the following callbacks: {list(difference)}. ", UserWarning
```

Comment:
```suggestion
                "Be aware that when using ``resume_from_checkpoint``,"
                " callbacks used to create the checkpoint need to be provided."
                f" Please add the following callbacks: {list(difference)}. ", UserWarning
```

Reply:
is this warning always needed? For example, i may have saved a checkpoint dict which contained callback states from the model checkpoint. But when I resume from checkpoint, I might not set a new model checkpoint callback on the trainer 

is there an issue/question which prompted the PR?

Reply:
This PR checks the `AU~B` between `save_callback_state` and `current_callback_state`.
Basically, if you trained with a `CustomCallback` and you don't provide it back when `resuming`, 
we won't be able to reload the state as this callback isn't provided. 
Therefore we should provide a warning that `resume` might not have the expected behaviour.

Reply:
`resume_from_checkpoint` is bringing lot of confusion as we expect exactly the same arguments to the Trainer as provided when training before.